controllers/nmstate_controller_test.go

```diff
@@ -227,3 +277,50 @@ func copyManifests(manifestsDir string) error {
 	}
 	return nil
 }
+
+func checkTolerationInList(toleration corev1.Toleration, tolerationList []corev1.Toleration) bool {
+	found := false
+	for _, currentToleration := range tolerationList {
+		if isSuperset(toleration, currentToleration) {
+			found = true
+			break
+		}
+	}
+	return found
+}
+
+// isSuperset checks whether ss tolerates a superset of t.
+func isSuperset(ss, t corev1.Toleration) bool {
+	if apiequality.Semantic.DeepEqual(&t, &ss) {
+		return true
+	}
+
+	if t.Key != ss.Key &&
+		// An empty key with Exists operator means match all keys & values.
+		(ss.Key != "" || ss.Operator != corev1.TolerationOpExists) {
```

Comment:
I think this would be way easier to comprehend if there was a function predicate like 
```
func isKeyMatching(a, b corev1.Toleration) bool {
  if a.Key == b.Key || (b.Key == "" && b.Operator == corev1.TolerationOpExists) {
    return true
  }
  return false
}
```

and here just check 
```
if !isKeyMatching(t, ss) {
  return false
}
```

Reply:
I follow the suggestion. Thank you.